client/signup/steps/passwordless/index.jsx

```diff
@@ -0,0 +1,261 @@
+
+import FormStateStore from 'lib/form-state';
+import createFormStore from 'lib/form-state/store';
+const { getFieldValue } = FormStateStore;
+
+export class PasswordlessStep extends Component {
+	static propTypes = {
+		flowName: PropTypes.string,
+		translate: PropTypes.func,
+	};
+
+	static defaultProps = {
+		translate: identity,
+	};
+
+	state = {
+		errorMessages: null,
+		noticeMessage: '',
+		noticeStatus: '',
+		submitting: false,
+		showVerificationCode: false,
+	};
+
+	UNSAFE_componentWillMount() {
+		this.formStore = createFormStore( {
+			syncInitialize: {
+				fieldNames: [ 'email', 'code' ],
+			},
+		} );
+	}
+
+	handleFieldChange = event => {
+		event.preventDefault();
+
+		this.setState( {
+			errorMessages: null,
+		} );
+
+		this.formStore.handleFieldChange( {
+			name: event.target.name,
+			value: event.target.value,
+		} );
+	};
+
+	createUser = event => {
+		event.preventDefault();
+		const data = { email: getFieldValue( this.formStore.get(), 'email' ) };
+
+		this.setState( {
+			submitting: true,
+		} );
+
+		createPasswordlessUser( this.handleUserCreationRequest, null, data );
+	};
+
+	handleUserCreationRequest = ( error, response ) => {
+		if ( error ) {
+			this.setState( {
+				errorMessages: [ error.message ],
+				submitting: false,
+			} );
+
+			return;
+		}
+
+		this.setState( {
+			errorMessages: null,
+			noticeMessage: response && response.message,
+			noticeStatus: response && response.warning ? 'is-warning' : 'is-info',
+			showVerificationCode: true,
+			submitting: false,
+		} );
+	};
+
+	verifyUser = event => {
+		event.preventDefault();
+		const data = {
+			email: getFieldValue( this.formStore.get(), 'email' ),
+			code: getFieldValue( this.formStore.get(), 'code' ),
+		};
+
+		this.setState( {
+			errorMessages: null,
+			submitting: true,
+		} );
+
+		verifyPasswordlessUser( this.handleUserVerificationRequest, null, data );
+	};
+
+	handleUserVerificationRequest = ( error, providedDependencies ) => {
+		if ( error ) {
+			this.setState( {
+				errorMessages: [ error && error.message ],
+				submitting: false,
+			} );
+
+			return;
+		}
+
+		this.submitStep( providedDependencies );
+	};
+
+	submitStep = providedDependencies => {
+		const { flowName, stepName } = this.props;
+
+		SignupActions.submitSignupStep(
+			{
+				flowName,
+				stepName,
+			},
+			providedDependencies
+		);
+
+		this.props.goToNextStep();
+	};
+
+	createUserButtonText() {
+		const { translate } = this.props;
+
+		if ( this.state.submitting ) {
+			return translate( 'Creating Your account…' );
+		}
+
+		return translate( 'Create your account' );
+	}
+
+	validateButtonText() {
+		const { translate } = this.props;
+
+		if ( this.state.submitting ) {
+			return translate( 'Verifying your code…' );
+		}
+
+		return translate( 'Verify your code' );
+	}
+
+	renderNotice() {
+		return (
+			this.state.noticeMessage && (
+				<Notice status={ this.state.noticeStatus }>{ this.state.noticeMessage }</Notice>
```

Comment:
We can't dismiss the notice at the moment:
![demo-unable-dismiss-notification](https://user-images.githubusercontent.com/1842898/58082709-1f58fc80-7bea-11e9-82ac-2aeb7dca9522.gif)

I personally prefer it to be not dismissable, given the way it is presented. We could add `isCompact` so that the dismiss button won't show up. Otherwise, adding a `onDismissClick()` callback would be helpful.